flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/serialization/DynamoDbStreamsDeserializationSchema.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.dynamodb.source.serialization;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+import org.apache.flink.connector.dynamodb.source.DynamoDbStreamsSource;
+import org.apache.flink.util.Collector;
+
+import software.amazon.awssdk.services.dynamodb.model.Record;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+/**
+ * This is a deserialization schema specific for the {@link DynamoDbStreamsSource}. Different from
+ * the basic {@link DeserializationSchema}, this schema offers additional DynamoDb-specific
+ * information about the record that may be useful to the user application.
+ *
+ * @param <T> The type created by the deserialization schema.
+ */
+@PublicEvolving
```

Comment:
Might be better to mark this as `@Experimental`

Reply:
marked as experimental